api/client/inspect.go

```diff
@@ -60,38 +62,37 @@ func (cli *DockerCli) CmdInspect(args ...string) error {
 
 	for _, name := range cmd.Args() {
 		if *inspectType == "" || *inspectType == "container" {
-			obj, _, err = readBody(cli.call("GET", "/containers/"+name+"/json?"+v.Encode(), nil, nil))
+			obj, statusCode, err = readBody(cli.call("GET", "/containers/"+name+"/json?"+v.Encode(), nil, nil))
 			if err != nil {
 				if err == errConnectionFailed {
 					return err
 				}
 				if *inspectType == "container" {
-					if strings.Contains(err.Error(), "No such") {
-						fmt.Fprintf(cli.err, "Error: No such container: %s\n", name)
-					} else {
+					if statusCode != http.StatusNotFound {
 						fmt.Fprintf(cli.err, "%s", err)
 					}
 					status = 1
+					fmt.Fprintf(cli.err, "Error: No such container: %s\n", name)
```

Comment:
Shouldn't this one be in a `else` ? (otherwise we're gonna print it for any error, which it does not now)